### PR TITLE
chore(medusa-core-utils): Migrate to TS

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CORE-836

### CODE REVIEW

File: packages/medusa-core-utils/src/compare-objects.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { differenceWith, isEqual } from "lodash"
```
Comment (srindom) [inline packages/medusa-core-utils/src/compare-objects.ts:1]: thought(this-file, non-blocking): is this used anywhere in the core? Can see it has some legacy usage of `_id` from the mongo times so suspect it isn't of much use any longer.

File: packages/medusa-core-utils/src/create-require-from-path.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import Module from "module"
+const Module = require("module")
 import path from "path"
 
-const fallback = filename => {
+const fallback = (filename: string) => {
   const mod = new Module(filename)
 
   mod.filename = filename
```
Comment (srindom) [inline packages/medusa-core-utils/src/create-require-from-path.ts:7]: question: is mixing `require` and `import` on purpose; if so why? if not please fix :)

**Review by srindom — APPROVED** [review]
LGTM w. comments

File: packages/medusa-core-utils/src/compare-objects.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa-core-utils/src/compare-objects.ts]: It seams to not be used, I removed it 👍

**Comment by olivermrbl** [comment]
@adrien2p Can I get you to include a changeset? :)

**Pull request merged**